gpio/imx6ull-gpio/imx6ull-gpio.c

```diff
@@ -255,10 +256,49 @@ void thread(void *arg)
 }
 
 
-int main(void)
+static int parse_args(int argc, char *argv[])
+{
+	int opt;
+	unsigned long prio;
+	char *endptr;
+
+	while ((opt = getopt(argc, argv, "p:")) != -1) {
+		switch (opt) {
+			case 'p':
+				prio = strtoul(optarg, &endptr, 0);
+				if (endptr == optarg) {
+					printf("%s: incorrect priority value (%s)\n", argv[0], optarg);
+					return -1;
+				}
+
+				if (prio > 7) {
+					printf("%s: incorrect priority value (%lu). It must be in [0,7] range\n",
+						argv[0], prio);
+					return -1;
+				}
+
+				priority(prio);
+				break;
+
+			default:
+				printf("gpio driver for imx6ull. Usage:\n");
+				printf("%s -p priority\n", argv[0]);
```

Comment:
nitpicking, as `-p priority` is an optional param, it should be in `[` `]` brackets